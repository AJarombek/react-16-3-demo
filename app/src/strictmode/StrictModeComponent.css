/**
 * Stylesheet for the StrictModeComponent & StrictModeChild components.
 * Author: Andrew Jarombek
 * Date: 1/27/2020
 */

.strict-mode-component {
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin: 50px 0;
}

.strict-mode-component div {
    margin: 0 10px;
    display: flex;
    align-items: center;

    width: 100px;
    height: 150px;
    text-align: center;
    border-radius: 2px;
}

.inside-strict-mode {
    background-color: aquamarine;
}

.outside-strict-mode {
    background-color: lightskyblue;
}
