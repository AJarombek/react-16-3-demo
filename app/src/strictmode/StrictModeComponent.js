/**
 * Demonstrate a component that operates in strict mode.
 * @author Andrew Jarombek
 * @since 1/26/2020
 */

import React from 'react';

const StrictModeComponent = () => {
  return (
    <React.StrictMode>
      <StrictModeChild />
    </React.StrictMode>
  );
};

class StrictModeChild extends React.Component {

  constructor(props) {
    super(props);
  }

  /**
   * Prove that StrictMode will complain about the legacy componentWillUpdate() method.
   */
  componentWillUpdate(nextProps, nextState) {
    console.info('Inside componentWillUpdate');
  }

  render() {
    return (
      <div className="in-strict-mode">

      </div>
    )
  }
}

export default StrictModeComponent;
