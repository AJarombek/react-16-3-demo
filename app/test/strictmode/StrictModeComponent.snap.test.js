/**
 * Snapshot test for the {@link ../../src/strictmode/StrictModeComponent} component.
 * @author Andrew Jarombek
 * @since 1/30/2020
 */

import React from 'react';
import StrictModeComponent from '../../src/strictmode/StrictModeComponent';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<StrictModeComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});
