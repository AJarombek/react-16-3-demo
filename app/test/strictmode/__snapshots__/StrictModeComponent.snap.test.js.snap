// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="strict-mode-component"
>
  <div
    className="outside-strict-mode"
  >
    <p>
      Outside Strict Mode
    </p>
  </div>
  <div
    className="inside-strict-mode"
  >
    <p>
      Inside Strict Mode
    </p>
  </div>
</div>
`;
